core/store/src/trie/state_snapshot.rs

```diff
@@ -83,32 +84,32 @@ impl StateSnapshot {
         store: TrieStoreAdapter,
         prev_block_hash: CryptoHash,
         flat_storage_manager: FlatStorageManager,
-        requested_shard_uids: &[ShardUId],
+        requested_shard_uids: &[(ShardIndex, ShardUId)],
         block: Option<&Block>,
     ) -> Self {
         tracing::debug!(target: "state_snapshot", ?requested_shard_uids, ?prev_block_hash, "new StateSnapshot");
         let mut included_shard_uids = vec![];
-        for shard_uid in requested_shard_uids {
-            if let Err(err) = flat_storage_manager.create_flat_storage_for_shard(*shard_uid) {
+        for &(shard_index, shard_uid) in requested_shard_uids {
+            if let Err(err) = flat_storage_manager.create_flat_storage_for_shard(shard_uid) {
                 tracing::warn!(target: "state_snapshot", ?err, ?shard_uid, "Failed to create a flat storage for snapshot shard");
```

Comment:
nit: add `shard_index` to log line

Reply:
I'm avoiding doing that for now because then I would have to do it everywhere and in vast majority of places it doesn't matter.